common/lib/xmodule/xmodule/js/src/video/09_video_caption.js

```diff
@@ -144,6 +172,139 @@ function (Sjson, AsyncProcess) {
             this.updatePlayTime(time);
         },
 
+        handleKeypressLink: function(event) {
+            var KEY = $.ui.keyCode,
+                keyCode = event.keyCode;
+
+            switch(keyCode) {
+                case KEY.UP:
+                    event.preventDefault();
+                    var focused = $(':focus').parent(),
+                        index = this.languageChooserEl.find('li').index(focused),
+                        total = this.languageChooserEl.find('li').size() - 1;
+
+                    this.previousLanguageMenuItem(event, index, total);
+                    break;
+
+                case KEY.DOWN:
+                    event.preventDefault();
+                    var focused = $(':focus').parent(),
+                        index = this.languageChooserEl.find('li').index(focused),
+                        total = this.languageChooserEl.find('li').size() - 1;
+
+                    this.nextLanguageMenuItem(event, index, total);
+                    break;
+
+                case KEY.ESCAPE:
+                    this.closeLanguageMenu(event);
+                    break;
+
+                case KEY.ENTER:
+                case KEY.SPACE:
+                    return true;
+            }
+        },
+
+        handleKeypress: function(event) {
+            var KEY = $.ui.keyCode,
+                keyCode = event.keyCode;
+
+            switch(keyCode) {
+                // Handle keypresses
+                case KEY.ENTER:
+                case KEY.SPACE:
+                case KEY.UP:
+                    event.preventDefault();
+                    this.openLanguageMenu(event);
+                    break;
+
+                case KEY.ESCAPE:
+                    this.closeLanguageMenu(event);
+                    break;
+            }
+
+            return event.keyCode === KEY.TAB;
+        },
+
+        nextLanguageMenuItem: function(event, index, total) {
+            event.preventDefault();
+
+            if (event.altKey) {
```

Comment:
Does it make sense to combine the event.altKey and event.shifKey conditions ? as in...
```
if (event.altKey || event.shiftKey) { return true; }
```